### DISCUSSION TITLE
How do I reinstall maestral without losing sync and indexing status? (switching WSL distros)

### DISCUSSION THREAD

**Discussion by lucas-figueiredo** [samschott/maestral#929] [Q&A] [answered]
I spent a couple of days trying to get the gui to work in Windows via WSL.
I used the info in #132 and #66 and got close, but that involved a whole gnome+ubuntu-desktop installation and subsequent removal, plus a couple of weird graphic apps from the Windows store.
Maestral CLI kept working through all that, and still works.
But now my WSL installation has a lot of cruft, and feels slow. And I don't have the chops to fix it manually.
I'd rather start fresh, but what's stopping me is the fact that the initial indexing of my 2TB dropbox took a few days on this crappy spinning disk, and it was loud.

The actual question is:
What are the files, configs, dbs, etc. I should backup and restore so that I keep my state between distros?

Thanks!

**Comment by samschott** [comment, marked as answer]
Wow, I admire your persistence in trying to get the GUI to work on WSL!

Backing up and restoring the following folders should prevent reindexing:

* `~/.config/maestral` for configuration
* `~/.local/share/maestral` for sync state
* The local Dropbox folder itself.
* Potentially the Dropbox token stored in your keychain (not sure which keychain is used in WSL).

If you skip the last step, you will need to relink Maestral after restoring all folders using `maestral auth link -r`.

**Reply by lucas-figueiredo** [reply]
Thanks! It took me a while to muster the courage to tackle this again, only got back to it a couple of days ago.
The migration worked perfectly.
I struggled with `maestral autostart -Y` because it wouldn't parse the config name correctly (it tried to name the service `maestral.daemon@%i.service`), but a manually adding a systemd unit seems to work.
I then started having problems with starting WSL at Windows logon and keeping it alive, but I'll spare you the details.

Only "problem" is that apparently every time I reboot the PC, Local indexing takes a while:
`sync DEBUG: Local indexing completed in 4052.3047 sec`.
And my disk usage during that time is effectively 0.
I'll try and get more info on _when_ the indexing happens (host machine reboot or VM reboot) during the day today, even if 1 hour at a time haha.

Thanks again! Love the product!

**Reply by lucas-figueiredo** [reply]
Oh, by the way, I think I figured out a way to have the GUI work in WSL, but it involves setting up an X server and accessing it via Remote Desktop, so I decided to just stay on the CLI and keep it light.
